core/src/main/java/org/wso2/carbon/kernel/utils/ManifestElement.java

```diff
@@ -0,0 +1,483 @@
+                while (c == ':') { // may not really be a :=
+                    c = tokenizer.getChar();
+                    if (c != '=') {
+                        String restOfNext = tokenizer.getToken("=:");
+                        if (restOfNext == null) {
+                            throw new Exception(MANIFEST_INVALID_HEADER_EXCEPTION + " Header: " + header + "Value: " +
+                                    value);
+                        }
+                        next = next.concat(":" + c + restOfNext);
+                        c = tokenizer.getChar();
+                    } else {
+                        directive = true;
+                    }
+                }
+                // determine if the attribute is the form attr:List<type>
+                String preserveEscapes = null;
+                String tempNextWithoutFirstLetter = next.substring(1);
+                if (!directive && tempNextWithoutFirstLetter.contains("List")) {
+                    Tokenizer listTokenizer = new Tokenizer(next);
+                    String attrKey = listTokenizer.getToken(":");
+                    if (attrKey != null && listTokenizer.getChar() == ':' && "List"
+                            .equals(listTokenizer.getToken("<"))) {
+                        // we assume we must preserve escapes for , and "
+                        preserveEscapes = "\\,";
+                    }
+                }
+
+                String val = tokenizer.getString(";,", preserveEscapes);
+                if (val == null) {
+                    throw new Exception(MANIFEST_INVALID_HEADER_EXCEPTION + " Header: " + header + "Value: " + value);
+                }
+
+                logger.debug(";" + next + "=" + val);
+                try {
+                    if (directive) {
+                        manifestElement.addDirective(next, val);
+                    } else {
+                        manifestElement.addAttribute(next, val);
+                    }
+                    directive = false;
+                } catch (Exception e) {
+                    throw new Exception(MANIFEST_INVALID_HEADER_EXCEPTION + " Header: " + header + "Value: " + value);
+                }
+                c = tokenizer.getChar();
+                if (c == ';') /* more */ {
+                    next = tokenizer.getToken("=:");
+                    if (next == null) {
+                        throw new Exception(MANIFEST_INVALID_HEADER_EXCEPTION + " Header: " + header + "Value: " +
+                                value);
+                    }
+                    c = tokenizer.getChar();
+                }
+            }
+            headerElements.add(manifestElement);
+            if (c == ',') { /* another manifest element */
+                continue;
+            }
+            if (c == '\0') { /* end of value */
+                break;
+            }
+            throw new Exception(MANIFEST_INVALID_HEADER_EXCEPTION + " Header: " + header + "Value: " + value);
```

Comment:
Here we have taken org.eclipse.osgi.util.ManifestElement class into our code base to get rid of unwanted dependencies. Thus the logic is intact from it's original version to make sure the code does not break.